pkg/webhook/preflight/preflight.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package preflight
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type (
+	// CheckerFactory returns a Checker for a given cluster.
+	CheckerFactory func(client ctrlclient.Client, cluster *clusterv1.Cluster) Checker
+	Checker        interface {
+		// Init returns the checks that should run for the cluster.
+		Init(ctx context.Context) []Check
+	}
+
+	Check       = func(ctx context.Context) CheckResult
```

Comment:
It would help to know the name of the Check, even if it fails to run (e.g. it panics). Maybe this should be an interface with two methods, `Run` and `Name`?

Reply:
If a check panics, the client may want to skip it, and doing that requires knowing its name.